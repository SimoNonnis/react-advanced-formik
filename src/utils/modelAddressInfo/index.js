function modelAddressInfo(list) {
  const [info] = list;

  return {
    sub_building: info.SubBuilding || '',
    building_name: info.BuildingName || '',
    building_number: info.BuildingNumber || '',
    street: info.PrimaryStreet || '',
    town: info.PostTown || '',
    county: info.County || '',
    postcode: info.Postcode || '',
  };
}

export default modelAddressInfo;
