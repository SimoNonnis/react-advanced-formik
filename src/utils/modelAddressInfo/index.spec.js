/* eslint-env jest */
import modelAddressInfo from './';

const mockPayload = [
  {
    Udprn: '52509479',
    Company: 'P C A Predict',
    Department: '',
    Line1: 'Waterside',
    Line2: 'Basin Road',
    Line3: '',
    Line4: '',
    Line5: '',
    PostTown: 'Worcester',
    County: 'Worcestershire',
    Postcode: 'WR5 3DA',
    Mailsort: '94142',
    Barcode: '(WR53DA1PX)',
    Type: 'SmallBusiness',
    DeliveryPointSuffix: '1P',
    SubBuilding: '',
    BuildingName: 'Waterside',
    BuildingNumber: '',
    PrimaryStreet: 'Basin Road',
    SecondaryStreet: '',
    DoubleDependentLocality: '',
    DependentLocality: '',
    PoBox: '',
    PrimaryStreetName: 'Basin',
    PrimaryStreetType: 'Road',
    SecondaryStreetName: '',
    SecondaryStreetType: '',
    CountryName: 'England',
    Location: { id: null, latitude: 52.1829787, longitude: -2.2216089 },
  },
];

describe('Test modelAddressInfo()', () => {
  it('should transform payload array to new object', () => {
    expect(modelAddressInfo(mockPayload)).toMatchSnapshot();
  });
});
