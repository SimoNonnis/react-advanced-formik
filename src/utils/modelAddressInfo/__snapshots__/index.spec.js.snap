// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test modelAddressInfo() should transform payload array to new object 1`] = `
Object {
  "building_name": "Waterside",
  "building_number": "",
  "county": "Worcestershire",
  "postcode": "WR5 3DA",
  "street": "Basin Road",
  "sub_building": "",
  "town": "Worcester",
}
`;
